.github/workflows/CodeQL Security Scan.yml

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: CodeQL Security Scan Pull request 
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+env:
+  INFORMATIONAL_VERSION: "6.0"
+  BUILD_INCREMENTER: "1"
+  CONFIGURATION: "Release"
+  CURRENT_BRANCH: ${{ GITHUB.HEAD_REF }}
+
+jobs:
+  analyze:
+    name: Analyze Code
+    runs-on: windows-latest
+    
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'csharp' ]
```

Comment:
Why is matrix needed here? I believe we can simply set ``languages: 'csharp'`` on the CodeQL initialization below.